src/query/fuzzy_query.rs

```diff
@@ -0,0 +1,144 @@
+use common::BitSet;
+use core::SegmentReader;
+use fst::Automaton;
+use levenshtein_automata::{LevenshteinAutomatonBuilder, DFA};
+use query::BitSetDocSet;
+use query::ConstScorer;
+use query::{Query, Scorer, Weight};
+use schema::{Field, IndexRecordOption, Term};
+use termdict::{TermDictionary, TermStreamer};
+use Result;
+use Searcher;
+
+/// A Fuzzy Query matches all of the documents
+/// containing a specific term that is with in
+/// Levenshtein distance
+#[derive(Debug, Clone)]
+pub struct FuzzyQuery {
+    term: Term,
+    distance: u8,
+}
+
+impl FuzzyQuery {
+    /// Creates a new Fuzzy Query
+    pub fn new(term: Term, distance: u8) -> FuzzyQuery {
```

Comment:
add options to handle 
- `Damerau` (that is consider transposition as having a cost of 1)
- `prefix queries`

Both are options available in the LevenshteinBuilder.